controllers/k8sgpt_controller.go

```diff
@@ -199,7 +206,15 @@ func (r *K8sGPTReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 			r.k8sGPTClients[k8sgptConfig.Name] = k8sgptClient
 		}
 
-		response, err := r.k8sGPTClients[k8sgptConfig.Name].ProcessAnalysis(deployment, k8sgptConfig)
+		if k8sgptConfig.Spec.RemoteCache != nil {
```

Comment:
This needs to be called once and only once right? Or do you think we should be updating the config pre-analysis? It might be a good thing I suppose

Reply:
Good point, I think it's necessary to call it in two cases: when the object is created and when it's modified. What do you think ?